main.tf

```
@@ -45,7 +46,7 @@ resource "aws_cloudwatch_log_group" "default" {
 
 resource "aws_iam_role_policy_attachment" "default" {
   provider   = aws.lambda
-  count      = var.role_arn == null && var.cloudwatch_logs ? 1 : 0
+  count      = local.create_policy == 1 && var.cloudwatch_logs ? 1 : 0
```

```suggestion
  count      = local.create_policy && var.cloudwatch_logs ? 1 : 0
```

This needs to be checked but it should be truthy if not `null` - @svanharmelen ?